Blog - WebApi v1.0.1 (#78)
arcus-azure/arcus: community/blog/2020-05-New-authorization-and-enhanced-logging-in-WebApi-v1.0.1/README.md

> # New authorization & enhanced logging in WebApi v1.0.1

Reviewer: Let's not focus on our fix :p
```suggestion
# New authorization & enhanced logging in WebApi v1.0
```

> With the release of version [`Arcus.WebApi v1.0.1`](https://github.com/arcus-azure/arcus.webapi/releases/tag/v1.0.1), the logging functionality is greatly enhanced for better tracing in your current and future applications.

Reviewer: ```suggestion
With the release of version [Arcus WebApi v1.0](https://github.com/arcus-azure/arcus.webapi/releases/tag/v1.0.1), the logging functionality is greatly enhanced for better tracing in your current and future applications.
```

* Determine the request header name you want to use where the access token should be available

```csharp
public void ConfigureServices()
{
    services.AddMvc(mvcOptions => mvcOptions.Filters.AddJwtTokenAuthorization());
}
```

For more information, see the [docs](https://webapi.arcus-azure.net/features/security/auth/jwt).

## Request tracking

Starting from this version, the library allows to track incoming requests with the use of a new middleware component to add to pipeline.

```csharp
public void Configure(IApplicationBuilder app, IWebHostEvironment env)
{
    app.UseRequestTracking();

    ...
    app.UseMvc();
}
```

By default the request body is ignored and the headers are logged (except for some default security headers that can be altered).

Example:

`HTTP Request POST http://localhost:5000/weatherforecast completed with 200 in 00:00:00.0191554 at 03/23/2020 10:12:55 +00:00 - (Context: [Content-Type, application/json], [Body, {"today":"cloudy"}])`

A lot more custom configuration and alterations can be made on this component.
For more information, see the [docs](https://webapi.arcus-azure.net/features/logging#logging-incoming-requests).

> ## Correlation merging

Reviewer: I would split features and breaking changes so that we have following "sections":
- New security feature
- New correlation feature
- Breaking changes

Author: Extracted and moved to a breaking changes section.

> # New authorization & enhanced logging in WebApi v1.0

Reviewer: How about this?
```suggestion
# Out-of-the-box request tracking, simplified HTTP correlation & JWT authorization in Arcus Web API v1.0
```

> With the release of version [Arcus WebApi v1.0](https://github.com/arcus-azure/arcus.webapi/releases/tag/v1.0.1), the logging functionality is greatly enhanced for better tracing in your current and future applications.

Reviewer: ```suggestion
We've shipped [Arcus Web API v1.0](https://github.com/arcus-azure/arcus.webapi/releases/tag/v1.0.1) which brings a couple of new features:
- Out-of-the-box Request tracking
- Making it easier to use HTTP correlation
- JWT Authorization
- Simplified NuGet package structure
```

Reviewer: Maybe also good to mention https://github.com/arcus-azure/arcus.webapi/issues/158

Come and take a look at what it has to offer!

## JWT authorization

> This version has so a [renamed component](https://github.com/arcus-azure/arcus.webapi/issues/149) in the new `Arcus.WebApi.Security` package called: JWT authorization.

Reviewer: ```suggestion
As of v1.0, we have renamed our existing `<name>` into `<>` because [we feel it's broader than just Managed Identity](https://github.com/arcus-azure/arcus.webapi/issues/149).
```

This component provides a mechanism that uses JWT (JSON Web Tokens) to authorize requests access to the web application.

This authorization process consists of the following parts:
* Find the OpenID server endpoint to request the correct access token
* Determine the request header name you want to use where the access token should be available

```csharp
public void ConfigureServices()
{
    services.AddMvc(mvcOptions => mvcOptions.Filters.AddJwtTokenAuthorization());
}
```

For more information, see the [docs](https://webapi.arcus-azure.net/features/security/auth/jwt).

> ## Request tracking

Reviewer: What about this?
```suggestion
## Out-of-the-box Request tracking
```

> ## Request tracking

Author: Done.

    app.UseRequestTracking();

    ...
    app.UseMvc();
}
```

By default the request body is ignored and the headers are logged (except for some default security headers that can be altered).

Example:

`HTTP Request POST http://localhost:5000/weatherforecast completed with 200 in 00:00:00.0191554 at 03/23/2020 10:12:55 +00:00 - (Context: [Content-Type, application/json], [Body, {"today":"cloudy"}])`

A lot more custom configuration and alterations can be made on this component.
For more information, see the [docs](https://webapi.arcus-azure.net/features/logging#logging-incoming-requests).

## Correlation merging

We have also made HTTP correlation with Serilog easier with the following extensions:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddHttpCorrelation();
}

public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
{
    app.UseHttpCorrelation();
    
    Log.Logger = new LoggerConfiguration()
        .Enrich.WithHttpCorrelationInfo()
        .WriteTo.Console()
        .CreateLogger();
}
```

For more information on correlation, see the [docs](https://webapi.arcus-azure.net/features/correlation).

## Breaking changes

> 

Reviewer: Maybe let's make it a bit more concise like this?

```suggestion

Last but not least, we have simplified our package structure to make it easier to keep track of dependencies.

Here is an overview:
- `Arcus.WebApi.Security.Authentication` and `Arcus.WebApi.Security.Authorization` are now consolidated into `Arcus.WebApi.Security`
- `Arcus.WebApi.Correlation` has been moved into `Arcus.WebApi.Logging`
```
